custom_components/tuya_local/devices/geyserwise_mws_water_heater.yaml

```diff
@@ -0,0 +1,217 @@
+name: Geyserwise MWS
+products:
+  - id: qwbyia2yr5ip9y4v
+    manufacturer: Geyserwise
+    model: MWS
+    name: Solar MPPT Geyser
+entities:
+  - entity: water_heater
+    dps:
+      - id: 1
+        type: boolean
+        name: operation_mode
+        mapping:
+          - dps_val: false
+            value: "Off"
+          - dps_val: true
+            value: "On"
+      - id: 2
+        type: string
+        name: away_mode
+        mapping:
+          - dps_val: "Timer"
+            value: false
+          - dps_val: "Holiday"
+            value: true
+      - id: 10
+        type: integer
+        name: current_temperature
+        readonly: true
+      - id: 102
+        type: integer
+        name: temperature
+        unit: C
+        range:
+          min: 30
+          max: 75
+  - entity: binary_sensor
+    name: Element
+    class: running
+    category: diagnostic
+    readonly: true
+    dps:
+      - id: 13
+        type: string
+        name: sensor
+        mapping:
+          - dps_val: "Off"
+            value: false
+          - dps_val: "On"
+            value: true
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 20
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+      - id: 20
+        type: bitfield
+        name: fault_code
+  - entity: sensor
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 20
+        type: bitfield
+        name: description
+        mapping:
+          - dps_val: 0
+            value: ok
+          - dps_val: 1
+            value: Over current
+          - dps_val: 2
+            value: Dry burn
+          - dps_val: 4
+            value: Wt sensor fault
+          - dps_val: 8
+            value: Heating loss
+          - dps_val: 16
+            value: Over temperature
+          - dps_val: 32
+            value: Water leak
+          - dps_val: 64
+            value: Comm failure
+          - dps_val: 128
+            value: Ct probe fault
+          - dps_val: 256
+            value: Pump failure
+  - entity: switch
+    name: AC Boost
```

Comment:
HA naming convention is to only use capitalisation on the first word of an entity name.